gc/2022/GC-08-23.md

```diff
@@ -33,3 +33,187 @@ Installation is required, see the calendar invite.
+
+BT: We are designing a format that has many different purposes, it’s a code format for interchange, we should prioritize all the tools that process that format, producers and consumers, and I don’t see how we justify that.. We could make the reverse argument, why aren’t there annotations that would be useful for tool authors and engines? From a different point of view that might be a hostile design decision
+
+AR: This is very real feedback from people that work in the research space, one of the top things that makes their life easier with Wasm was that there is no overload, which is the trivial case of the problem we’re discussing right now. We’re talking about a much more inconvenient relaxation.. I would also like to push back on the size arguments, it falls into the category of premature optimization, we haven’t really optimized for size as much previously, putting in adhoc solutions to individual small things, we could have had a const 0 inaructions, but we decided against it, we don’t have a negate instructions, and all these things we never decided to size optimized, we should do a more targeted approach instead of adhoc ideas, we should stay consistent with that
+
+BT: I think we would call this layer 1 compression, I see what we’re doing  as we’re trying to use the type reconstruction algorithm as a way to compression, if we really want to use this, that might be really cool. That’s a compression mechanism that might not be available to a general purpose zip algorithm, if we were to continue looking at that idea, that could exist at another level, and we can have another tool that decompresses, still get the best of both worlds
+
+AZ: Fair point that the compression could be done at another level, we haven’t focused on micro optimizing things. Want to push back against some of these, while in theory we could do something in the space, but it would take time, it’s unclear that we would actually do this. Unclear that we would actually have the resources to work on this, there are things we could have done the past, but did not but those are things  (const 0, const 1) that compress well, on the other hand, type annotations are high entropy thing and have a higher cost, this stuff does matter, we have a lot of use cases on teh web where there are limited on the size of the download, while it doesn’t matter in some countries with unlimited bandwidth, other countries it isn’t. 5% is significant, lots of efforts to reduce code size overall and ever % matters. IT is likely that if we don’t save this now, it’s possible that we’ll be stuck with this for a very long time and that would be wasteful.
```

Comment:
```suggestion
AZ: Fair point that the compression could be done at another level, we haven’t focused on micro optimizing things. Want to push back against some of these, while in theory we could do something in the space, but it would take time, it’s unclear that we would actually do this. Unclear that we would actually have the resources to work on this, there are things we could have done the past, but did not but those are things  (const 0, const 1) that compress well, on the other hand, type annotations are high entropy thing and have a higher cost, this stuff does matter, we have a lot of use cases on the web where there are limited on the size of the download, while it doesn’t matter in some countries with unlimited bandwidth, other countries it isn’t. 5% is significant, lots of efforts to reduce code size overall and ever % matters. It is likely that if we don’t save this now, it’s possible that we’ll be stuck with this for a very long time and that would be wasteful.
```